sem/database.py

```diff
@@ -264,12 +291,16 @@ def insert_result(self, result):
         # Insert result
         self.db.table('results').insert(deepcopy(result))
 
-    def get_results(self, params=None, result_id=None):
+    def get_results(self, params=None, result_id=None, log_component=None):
         """
         Return all the results available from the database that fulfill some
         parameter combinations.
 
-        If params is None (or not specified), return all results.
+        If params is None (or not specified) and log_component is specified,
+        return all results where the specified log_component is enabled.
+
+        If params is None (or not specified) and log_component is None 
+        (or not specified), return all results with logging disabled.
```

Comment:
I agree we should support all three cases. However, I would also like the default behavior to be coherent with the current behavior, i.e., only results with no logging enabled are returned. If the user wants to get all results, including those with logging (this is a corner case, in my opinion), they should be able to do so - I agree that `None` is a decent option for this behavior. Not very intuitive, but decent.

At this point, to preserve SEM's current behavior, we should use `{}` as a default value for `log_components`. However, to me this suggests that, by default, we want logging (`None` is the better option to express that you are not interested in logging).

Might need to discuss this in the next meeting.

Reply:
> At this point, to preserve SEM's current behavior, we should use `{}` as a default value for `log_components`. However, to me this suggests that, by default, we want logging (`None` is the better option to express that you are not interested in logging).

I agree `None` will be the better choice to represent that the user is not interested in logging. To accommodate this, what if we modify the conditions as follows: 

- all results, regardless of log components (both enabled and disabled) if `log_component` is an empty dictionary
- only results with specified log component if `log_component` is non-empty dict
- only results with no log components if `log_component` is None


Note: These conditions will be in addition to the conditions on the parameters passed.

Reply:
I agree, the default behavior should not be changed.

> At this point, to preserve SEM's current behavior, we should use `{}` as a default value for `log_components`. However, to me this suggests that, by default, we want logging (`None` is the better option to express that you are not interested in logging).

Using `{}` as default value is actually highly unrecommended (check [here](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)).

I'm ok with Akshit's proposal, but it should be done consistently for all functions!

Reply:
I'm ok with Akshit's proposal, too! But we need to document this extremely clearly in the docstring, since it's not very intuitive what `{}` should do at a first glance.